hastory-data-gen/src/Data/Hastory/Gen.hs

```diff
@@ -29,11 +30,13 @@ instance GenValid UserForm where
   shrinkValid = shrinkValidStructurally
 
 instance GenValid Username where
-  genValid = Username <$> validUsernameText
+  genValid = Username <$> userNameTextGen
     where
-      validUsernameText = T.pack . map unUsernameChar <$> vectorOf 4 genValid
-  shrinkValid = shrinkValidStructurally
-
-instance GenValid UsernameChar where
-  genValid = UsernameChar <$> choose ('\0', '\127') `suchThat` isValid
+      userNameTextGen = oneof [lengthFourText, arbitraryLengthText]
```

Comment:
nono, you want to add them together, and use length three:

```
liftA2 (<>) lengthThreeText arbitraryLengthText
```

Reply:
Ah - I see. This solution didn't come to me because I only had a "not null" check on the username's length. I've now changed the instance of `Validity Username` to reflect that it must be greater than 2 chars.